xchange-stream-coinbasepro/src/main/java/info/bitrich/xchangestream/coinbasepro/CoinbaseProStreamingExchange.java

```diff
@@ -84,11 +84,17 @@ public Completable disconnect() {
 
   @Override
   public Observable<Throwable> reconnectFailure() {
+    if (streamingService == null)
+      return Observable.error(new NullPointerException("streamingService is null"));
```

Comment:
How about an `IllegalStateException("Not connected")`? More descriptive.

Reply:
works for me too